Summer 2020 updates

Daehoon7992 opened the pull request:
ui.r
1) Changed the format of the app.

2) Added space after each comma.

3) Added a reference tab and references

server.r
1) Streamlined the code

2) Removed unnecessary spaces

Daehoon7992 committed:
Summer 2020 updates

ui.r
1) Changed the format of the app.

2) Added space after each comma.

3) Added a reference tab and references

server.r
1) Streamlined the code

2) Removed unnecessary spaces

neilhatfield reviewed (changes requested):
Let's do the following changes:

1. Convert to the app.R structure rather than the ui.R/server.R structure. Use the following order in the new app.R:
    a. Load packages
    b. Add the app meta data lines
    c. Define any global constants and functions, load data sets.
    d. Create the UI; that is, `ui <- list( [code] )`
    e. Create the server; that is, `server <- function(input, output, session) { [code] }`
    f. Boast App Call: `boastUtils::boastApp(ui = ui, server = server)`
2. Go ahead and remove any packages that aren't needed: You had shinyjs and shinyDND commented out. I was able to get the app to run without MASS and DAAG as well. Amendment: I just came across pulling a data set out of MASS
3. With the boastApp call, you may remove the call to the boast.CSS file (lines 15-18 of the ui.R)
4. Add in the feedback button in the header
5. Change `icon("dashboard")` to `icon("tachometer-alt")`
6. The prereq's page isn't prereq's, rather it feels like instructions. I'm not sure that this app actually needs prereq's.
7. I'm unclear as to the point of the show histograms button. Why can't the histograms already be showing?
8. Explore the `ggExtra` package and the function `ggMarginal`. These can allow you to put the histograms as margins of the scatterplot. I think this would be a better approach to the current combination of three plots.
9. I feel like the data set selector should be up higher (above the well panel). This is because I think that the choice of data set should be visually before any explanation/description of the data set.
10. This goes with 8, but I would encourage you to use `ggplot2` rather than base graphics. If you want to stick with base, you must change the points to be solid rather than the hollow circles that they currently are (and you'll here me constantly suggest that you should use `ggplot2`).
11. Feel free to add more data sets/replace what's there.
12. I would go ahead and remove the "None" option for the data sets and just start with a data set.
13. All plots needs alt text (i.e., aria-labels).
14. You jump from h2 to h4 on the Explore page. That's a Skipped Heading error (no h3). You need to use headings sequentially. Remember, headings aren't for styling; they are for structuring.
15. I don't believe the popovers off anything; remove.
16. I don't think we need to offer a download of the data. Include the data in there references page so that if they want to get the data, we've pointed them to it.
17. Code formatting: You've done a decent job avoiding long code lines but you still need to go back through the code to 1) make arguments explicit, and 2) improve readability. Here's an example of what's currently in the code:
    ```
        selectInput("inputs", "Select Data Set", choices = c('None',
                                      'Animals', 'Earthquakes', "Countries")),

Here's a way to improve it:
```
selectInput(
  inputId = "dataSelected", #Notice I changed the name to something more informative
  label = "Select a data set",
  choices = c(
    "Animals", #This portion could be written as a single line
    "Earthquaks",
    "Countries"
  ),
)
``` 
18. Also, let's try to cut back on redundant coding. The app currently uses a conditional panel to swap out the horizontal/vertical axis controls depending on which data set is selected. This means that instead of have 2 controls, you have 6 controls. That's 3 times as many inputs as you need to keep track of. Instead, use just two--one for the horizontal axis, one for the vertical. Then use the `updateSelectInput` function to change the offerings/choices for both controls.
19. Remove any code no longer being used/commented out. Remove any unhelpful comments. For chunks like the "task" and "game", I think that we should go ahead and move that code out of the file. You can put it into an extra text file (e.g., futureIdeas.txt) which you can the upload with your next push. You can also make an issue with the enhancement tag that references that file so that next year's students can see it.
20. No colon necessary after "Instructions" in the Sweet Alert. Also, remove `btn_colors = "green"`.
21. Coding: only use = to assign a value to an argument. Use <- for all assignments including things like `output$animalPlot <- renderPlot({...` Note: renderPlot should be on the same line as output$animalPlot
22. See prior comment about reducing redundant code in relation to plots. Essentially, my preference would be that there is one plot which shows the scatterplot and histograms. The single plot would then listen to the controls and update as necessary.
23. What if instead of `checkboxGroupInput` for transforming/showing the transformations, you use `shinyWidgets::switchInput`instead?
24. I'm noticing the plot code that lines and legends were turned off. Any idea as to why?

I'm going to stop here for now so that you can start processing some of these changes.

Daehoon7992 committed:
Updates

1. Merged ui.R and server.R
2. Removed packages: shinyjs, shinyDND, MASS, and DAAG
3. Removed the call to the boast.CSS file
4. Added in the feedback button in the header
5. Changed icon("dashboard") to icon("tachometer-alt")
6. Removed the prerequisites page
7. Removed the histogram button and show without button
8. Moved the location of dataset selector to the higher position
9. Removed 'none' option in dataset selector and start with 'Animal' dataset
10. Changed 'h4' to 'p' on explore page
11. Removed unnecessary popovers
12. Removed download function
13. Revised the code formatting made explicit on UI section
14. Removed commented out codes
15. Removed colons
16. Conver '=' sign to '<-' sign

paridhi1603 committed:
Tried ggplot

paridhi1603 committed:
Chnanged to ggplot

paridhi1603 committed:
References added for Animal dataset

paridhi1603 committed:
Updated References

paridhi1603 commented:
- Changed from basic to ggplot
- Changed axis names
- Made headlines format correct

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1382 @@
+        menuItem("Explore", tabName = "explore", icon = icon("wpexplorer")), 
+        menuItem("References", tabName = "References", icon = icon("leanpub"))
+      ), 
+      tags$div(
```
Change logo to the newer one in the style guide

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1382 @@
+              ), 
+              br(), 
+              br(), 
+              h2("Acknowledgement"), 
```
Make sure to add your name to the acknowledgements as well as the updated by statement

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1382 @@
+                      inputId = "inputs",
+                      label = "Select Data Set",
+                      choices = c('Animals', 'Earthquakes', "Countries")
+                    ) 
```
Whenever I select the Earthquakes data set it says magnitude not found. I thought this was fixed in the meeting so maybe you forgot to push your code?

shravanisamala commented on app.R:
```diff
@@ -0,0 +1,1382 @@
+        class = "hangingindent",
+        "Joyner, W. B., Boore, D. M., and Porcella, R. D. (1981), 
+        Peak Horizontal Acceleration and Velocity from Strong-Motion Records Including Records from the 1979 Imperial Valley California Earthquake"
+      )
```
I believe there is a copyright insert from the style guide that is missing here that can be found on the app template.

shravanisamala reviewed (changes requested):
Overall looks pretty good! I think if you get a go ahead from the professors you can remove the commented out chunks of code. Also, the repo is missing a Description file so the console is creating a bunch of "unable to create XAPI statements"

Jiayue-He reviewed (commented):
I think LICENSE and README files should be stored in the docs folder.

paridhi1603 committed:
Update app.R

neilhatfield committed:
Partial Update

Partial update just need to resolve errors triggered by switching data sets

neilhatfield committed:
Update app.R

neilhatfield committed:
Final Fall 2021 Update

Final updates (baring typos and bugs) for Fall 2021

neilhatfield committed:
Removing Extra Files

delete extra files

neilhatfield committed:
Merge branch 'master' into Demo_DG

neilhatfield committed:
Merge branch 'Demo_DG'

neilhatfield committed:
Revert "Merge branch 'master' into Demo_DG"

This reverts commit c238d6adf4e2c3169d503e9a610def0e2ac6cd80, reversing
changes made to e72e315f11812cf202ba625045d3d7c5386fa85f.

neilhatfield committed:
Merge branch 'master' into Demo_DG

neilhatfield committed:
Create CODEOWNERS
